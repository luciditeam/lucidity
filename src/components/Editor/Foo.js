import React from 'react'
import {connect} from 'cerebral/react'
import {Computed} from 'cerebral'

const Name = Computed(
  {
    name: 'foo.bar'
  },
  function Name ({name}) {
    return name
  }
)

export default connect(
  {
    name: Name
  },
  function Foo ({name}) {
    return <div>{name}</div>
  }
)
