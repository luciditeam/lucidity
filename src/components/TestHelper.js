import React from 'react'
import {StateContainer} from 'cerebral/react'
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'

export function snapshot (block, state) {
  const wrapper = mount(
   <StateContainer state={state}>
      <div id='testhelper'>
       {block}
     </div>
   </StateContainer>
 )

  return mountToJson(wrapper.find('#testhelper').children())
}
