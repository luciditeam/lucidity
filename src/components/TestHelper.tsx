import { JSX } from './Component'
import { StateContainer } from 'cerebral/react'
import { mount } from 'enzyme'
export { mountToJson as snapshot } from 'enzyme-to-json'

export function render ( block: any, state?: any ) {
  const wrapper = mount
  ( <StateContainer state={ state }>
      <div id='testhelper'>
       { block }
     </div>
   </StateContainer>
 )
  return wrapper.find ( '#testhelper' ).children ()
}
