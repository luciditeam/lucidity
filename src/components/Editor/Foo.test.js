/* global it expect describe */
import React from 'react'
import {snapshot} from '../TestHelper'
import Foo from './Foo'

describe('Foo', () => {
  it('renders foo name', () => {
    const tree = snapshot(<Foo />, {foo: {bar: 'baz'}})
    expect(tree).toMatchSnapshot()
  })
})
