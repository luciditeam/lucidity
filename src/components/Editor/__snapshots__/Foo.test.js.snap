exports[`Foo renders foo name 1`] = `
<CerebralWrapping_Foo>
  <Foo
    name="baz">
    <div>
      baz
    </div>
  </Foo>
</CerebralWrapping_Foo>
`;
